Add description on how to configure KnativeServing for Istio (#4278)
knative/docs: docs/admin/install/knative-with-operators.md

layer, if the ingress is not specified in the Knative Serving CR. Click on each of the following tabs to see how you can configure
Knative Serving with different ingresses:

=== "Istio (default)"

    The following commands install Istio and enable its Knative integration.

    1. [Install Istio](installing-istio.md).

    1. To configure Knative Serving to use Istio, copy the following YAML into a file:
        ```yaml
        apiVersion: operator.knative.dev/v1alpha1
        kind: KnativeServing
        metadata:
          name: knative-serving
          namespace: knative-serving
        ```

    1. If Istio is installed under a namespace other than the default `istio-system`, add the `spec.config.istio` field
       to the YAML file as follows:
        ```yaml
        apiVersion: operator.knative.dev/v1alpha1
        kind: KnativeServing
        metadata:
          name: knative-serving
          namespace: knative-serving
        spec:
          config:
            istio:
              local-gateway.<local-gateway-namespace>.knative-local-gateway: "knative-local-gateway.<istio-namespace>.svc.cluster.local"
        ```

>         Where:
>         - `<local-gateway-namespace>` is the local gateway namespace, which is the same as Knative Serving namespace `knative-serving`.
>         - `<istio-namespace>` is the namespace where Istio is installed.

Reviewer: ```suggestion
        Where:
        
        - `<local-gateway-namespace>` is the local gateway namespace, which is the same as Knative Serving namespace `knative-serving`.
        - `<istio-namespace>` is the namespace where Istio is installed.
```

Reviewer: The formatting is broken here (see screenshot below). I've added an extra line before the bullet points to format it correctly.

<img width="827" alt="Screen Shot 2021-10-06 at 18 57 44" src="https://user-images.githubusercontent.com/33425661/136257918-937773bd-697d-49e8-9dad-17031135b7ba.png">

> ### Installing with different networking layers

Reviewer: ```suggestion
### Installing Knative Serving with different networking layers
```

> !!! note "Installing the Knative Serving component with different network layers"

Reviewer: ```suggestion
```

Reviewer: I don't think this whole section needs to be in a note.

    Knative Operator can configure Knative Serving component with different network layer options. Istio is the default network
    layer, if the ingress is not specified in the Knative Serving CR. Click on each of the following tabs to see how you can configure
    Knative Serving with different ingresses:

    === "Istio (default)"

        The following commands install Istio and enable its Knative integration.

        1. [Install Istio](installing-istio.md).

        1. To configure Knative Serving to use Istio, copy the following YAML into a file:
          ```yaml
          apiVersion: operator.knative.dev/v1alpha1
          kind: KnativeServing
          metadata:
            name: knative-serving
            namespace: knative-serving
          ```

>         1. If Istio is installed under a namespace other than the default `istio-system`, you need to change the content
>            of the YAML into:

Reviewer: ```suggestion
        1. If Istio is installed under a namespace other than the default `istio-system`, add the `spec` field
        to the YAML file as follows:
```

>           ```bash

Reviewer 2: ```suggestion
          ```yaml
```

          apiVersion: operator.knative.dev/v1alpha1
          kind: KnativeServing
          metadata:
            name: knative-serving
            namespace: knative-serving
          spec:
            config:
              istio:
                local-gateway.<local-gateway-namespace>.knative-local-gateway: "knative-local-gateway.<istio-namespace>.svc.cluster.local"
          ```

>           Where `<local-gateway-namespace>` is the local gateway namespace, which is the same as Knative Serving
>           namespace `knative-serving`, and <istio-namespace> is the namespace, where Istio is installed.

Reviewer: ```suggestion
              Where:
          
              - `<local-gateway-namespace>` is the local gateway namespace, which is the same as Knative Serving namespace `knative-serving`.
              - `<istio-namespace>` is the namespace where Istio is installed.
```

Reviewer: I'm not clear about what line 239 means:
> ... the local gateway namespace, which is the same as Knative Serving namespace `knative-serving`

Does this mean that `<local-gateway-namespace>` will always be `knative-serving`?

Author: <local-gateway-namespace> should be the same as the namespace for the knative serving. for knative serving installed with the operator, it is defaulted to knative-serving.

        1. Apply the YAML file by running the command:
          ```bash
          kubectl apply -f <filename>.yaml
          ```

>           Where `<filename>` is the name of the file you created in the previous step.

Reviewer: ```suggestion
              Where `<filename>` is the name of the file you created in the previous step.
```

Reviewer: Fixing formatting.

>         1. Fetch the External IP or CNAME:

Reviewer: ```suggestion
        1. Fetch the External IP or CNAME by running the command:
```

          ```bash
          kubectl get svc istio-ingressgateway -n <istio-namespace>

>           ```
> 
>           Save this for configuring DNS later.

Reviewer: ```suggestion
          ```

              Save this for configuring DNS later.
```

Reviewer: Fixing formatting.
